solidity/dashboard/src/services/rewards.service.js

```diff
@@ -13,26 +14,32 @@ const fetchAvailableRewards = async (web3Context) => {
   } = web3Context
   try {
     let totalRewardsBalance = web3Utils.toBN(0)
-    const expiredGroupsCount = await keepRandomBeaconOperatorContract.methods
-      .getFirstActiveGroupIndex()
-      .call()
-    const acitveGroups = await keepRandomBeaconOperatorContract.methods
-      .numberOfGroups()
-      .call()
-    const allGroups = add(expiredGroupsCount, acitveGroups).toNumber()
-    const groups = []
+    const operatorEventsSearchFilters = {
+      fromBlock: CONTRACT_DEPLOY_BLOCK_NUMBER[OPERATOR_CONTRACT_NAME],
+    }
+
+    // get all created groups
+    const groups = await contractService.getPastEvents(
+      web3Context,
+      OPERATOR_CONTRACT_NAME,
+      "DkgResultSubmittedEvent",
+      operatorEventsSearchFilters
+    )
```

Comment:
thoughts about filtering return data a bit here to make a `for` loop below cleaner? Similar to what we do in `tbt-authorization.service.js`
```
.map((_) => _.returnValues. groupPubKey)
```

Reply:
I think with `map` would be more complicated due to async calls and in some cases, we skip iteration with `continue` which is not possible with `map`.

Reply:
It shouldn't affect skipping the iteration. If we create a new array by extracting public keys this way:
```
const groupPubKeys = (
    await contractService.getPastEvents(
      web3Context,
      OPERATOR_CONTRACT_NAME,
      "DkgResultSubmittedEvent",
      operatorEventsSearchFilters
    )
).map((_) => _.returnValues.groupPubKey)
```
then below, we can just write a simple for loop:
```
for (let i = 0; i < groupPubKeys.length; i++) {
 // groupPubKeys[i]
}
```